design/dpop.md

```diff
@@ -0,0 +1,308 @@
+# OAuth 2.0 Demonstrating Proof-of-Possession at the Application Layer (DPoP)
+
+* **Status**: Notes
+* **JIRA**: [KEYCLOAK-15169](https://issues.redhat.com/browse/KEYCLOAK-15169)
+
+## Motivation
+
+DPoP is a mechanism to prevent illegal API calls from succeeding only with a stolen access token. This is done via augmenting the calls to OAuth 2.0 endpoints as well as resource servers with a special DPoP header that proves possession of a private key, and binding the issued tokens to that key. Please refer to the Resources section for more info on DPoP.    
+
+## Implementation Details
+
+DPoP is a cross-cutting concept that would affect multiple Keycloak components, namely core libraries, OAuth/OIDC endpoints, adapters, models and admin UI.
+
+It is suggested that DPoP should be set up on a per-client basis, with the three options available that should be interpreted as follows:
```

Comment:
One consideration is, whether to use client-specific configuration options or rather rely on client policies? It will be nice if we can move much of the client-specific configuration options from clients and drive them with the use of client policies. 

Especially for DPoP it can be nice to do that due the fact that DPoP specification does not seem to introduce any client-specific metadata for client registration?

@tnorimat WDYT?

Reply:
@mposolda Yes, as you pointed out, the current [DPoP draft(v3)](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-dpop-03) does not introduce any client metadata. Therefore, it might be better to rely on client policies and not introduce a new client setting for DPoP.